### PR TITLE
feat(Crave): add presence

### PR DESCRIPTION
**Pull request by Averyyyyyyyy**
## Acknowledgements
- [x] I read the [Presence Guidelines](https://github.com/PreMiD/Presences/blob/main/.github/CONTRIBUTING.md)
- [x] I linted the code by running `yarn format`
- [X] The PR title follows the repo's [commit conventions](https://github.com/PreMiD/Presences/blob/main/.github/COMMIT_CONVENTION.md)

## Screenshots
<details>
<summary> Proof showing the creation/modification is working as expected </summary>

Presence while on homepage
Presence while watching The King's Man
Presence while watching Euphoria

</details>

### CODE REVIEW

**Comment by Averyyyyyyyy** [comment]
so is this going to get reviewed 🥺

File: websites/C/Crave/presence.ts [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+		smallImageText: isVideo ? (isPlaying() ? "Playing" : "Paused") : "Browsing",
+		details: detailsString,
+		state: stateString,
+		startTimestamp: isVideo ? (isPlaying() ? getTimeStamps()[0] : 0) : null,
+		endTimestamp: isVideo ? (isPlaying() ? getTimeStamps()[1] : 0) : null,
+		buttons: isVideo
+			? [
+					{
+						label: "Watch",
+						url: document.URL,
```
Comment (EncryptedDev) [inline websites/C/Crave/presence.ts:150]:
```suggestion
						url: document.location.href,
```

File: websites/C/Crave/presence.ts [hunk 2]
```diff
@@ -0,0 +1,158 @@
+const presence = new Presence({
+	//The client ID of the Application created at https://discordapp.com/developers/applications
```
Comment (EncryptedDev) [inline websites/C/Crave/presence.ts:2]:
```suggestion
```

File: websites/C/Crave/presence.ts [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+				}
+			} else if (state === "website") return ["Browsing", "Website", ""];
+			else {
+				presence.error(`Unknown state: ${state}`);
+				return ["Error", "Unknown State", ""];
+			}
+	}
+}
+
+// Presence
```
Comment (EncryptedDev) [inline websites/C/Crave/presence.ts:115]:
```suggestion
```

**Review by EncryptedDev — COMMENTED** [review]
Hiya 👋🏼

Thank you for contributing to the presence store. I've just noticed a few things found below
- Please use JSDoc where needed and not on simple functions as you've done here
- We would much prefer if you were to create the presences by using if statements and elongate your code (use fewer functions) as currently it is very messy and hard to understand

Drop me a DM if you need some assistance fulfilling this.

**Comment by Averyyyyyyyy** [comment]
also you might want to label the PR with paid service

**Comment by Averyyyyyyyy** [comment]
#5284

File: websites/C/Crave/presence.ts [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+	browsingStamp = Date.now();
+
+function $(selector: string): HTMLElement {
+	return document.querySelector(selector);
+}
+
+presence.on("UpdateData", async () => {
+	const presenceData: PresenceData = { largeImageKey: "crave_logo" };
+
+	if ($(".jw-video") !== null) {
```
Comment (EncryptedDev) [inline websites/C/Crave/presence.ts:13]:
```suggestion
	if ($(".jw-video")) {
```

File: websites/C/Crave/presence.ts [hunk 5]
```diff
@@ -0,0 +1,67 @@
+const presence = new Presence({
+		clientId: "1001288215388495953",
+	}),
+	browsingStamp = Date.now();
+
+function $(selector: string): HTMLElement {
+	return document.querySelector(selector);
+}
```
Comment (EncryptedDev) [inline websites/C/Crave/presence.ts:8]: Please do not use a sudo function like so, you can just explicitly write the querySelector at each reference

File: websites/C/Crave/presence.ts [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+presence.on("UpdateData", async () => {
+	const presenceData: PresenceData = { largeImageKey: "crave_logo" };
+
+	if ($(".jw-video") !== null) {
+		// if contains video
+		if ($(".jw-icon-playback").ariaLabel !== "Play") {
+			// video is playing
+
+			const elapsed = presence.timestampFromFormat(
+					$(".jw-text-elapsed").innerHTML
```
Comment (EncryptedDev) [inline websites/C/Crave/presence.ts:19]: This is preferable
```suggestion
					$(".jw-text-elapsed").textContent
```

File: websites/C/Crave/presence.ts [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+		}
+
+		presenceData.buttons = [
+			{
+				label: "Watch",
+				url: document.location.href,
+			},
+		];
+
+		if ($(".bm-icon-episode-list") !== null) {
```
Comment (EncryptedDev) [inline websites/C/Crave/presence.ts:45]:
```suggestion
		if ($(".bm-icon-episode-list")) {
```

**Review by EncryptedDev — CHANGES_REQUESTED** [review]

**Review by EncryptedDev — APPROVED** [review]

**Review by Bas950 — APPROVED** [review]

**Pull request merged**